pkg/git/repository/connectivity.go

```diff
@@ -0,0 +1,71 @@
+package repository
+
+import (
+	"fmt"
+	"github.com/redhat-developer/devconsole-git/pkg/log"
+	gittransport "gopkg.in/src-d/go-git.v4/plumbing/transport"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+const (
+	unableReachRepoError   = "unable to reach the repo"
+	unableFindBranchError  = "unable to find the branch"
+	unableParseBranchError = "unable to parse the branch"
+)
+
+func IsReachableWithBranch(log *log.GitSourceLogger, branch string, endpoint *gittransport.Endpoint) (bool, error) {
```

Comment:
could we please add a func comment here describing what the function does?

Reply:
added here 919a1a0